src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -2030,15 +2053,23 @@ UpdateShardMoveDependencies(PlacementUpdateEvent *move, uint64 colocationId, int
 		shardMoveDependencies.colocationDependencies, &colocationId, HASH_ENTER, NULL);
 	shardMoveDependencyInfo->taskId = taskId;
 
-	shardMoveDependencyInfo = hash_search(shardMoveDependencies.nodeDependencies,
-										  &move->sourceNode->nodeId, HASH_ENTER, NULL);
-
-	shardMoveDependencyInfo->taskId = taskId;
-
-	shardMoveDependencyInfo = hash_search(shardMoveDependencies.nodeDependencies,
-										  &move->targetNode->nodeId, HASH_ENTER, NULL);
-
-	shardMoveDependencyInfo->taskId = taskId;
+	bool found;
+	ShardMoveSourceNodeHashEntry *shardMoveSourceNodeHashEntry = hash_search(
+		shardMoveDependencies.nodeDependencies, &move->sourceNode->nodeId, HASH_ENTER,
+		&found);
+	if (found)
+	{
+		int64 *newTaskId = palloc0(sizeof(int64));
+		*newTaskId = taskId;
+		shardMoveSourceNodeHashEntry->taskIds = lappend(
+			shardMoveSourceNodeHashEntry->taskIds, newTaskId);
+	}
+	else
+	{
+		int64 *firstTaskId = palloc0(sizeof(int64));
+		*firstTaskId = taskId;
+		shardMoveSourceNodeHashEntry->taskIds = list_make1(firstTaskId);
+	}
```

Comment:
nit: I think this looks a bit simpler:
```suggestion
	if (!found)
	{
		shardMoveSourceNodeHashEntry->taskIds = NIL;
	}
	int64 *newTaskId = palloc0(sizeof(int64));
	*newTaskId = taskId;
	shardMoveSourceNodeHashEntry->taskIds = lappend(
			shardMoveSourceNodeHashEntry->taskIds, newTaskId);
```